src/core/lightning/payment-status-checker.ts

```diff
@@ -0,0 +1,28 @@
+import { LnInvoiceLookupError } from "@domain/errors"
+import { decodeInvoice } from "@domain/ln-invoice"
+import { InvoiceUser } from "@services/mongoose/schema"
+
+const PaymentStatusChecker = ({ paymentRequest, lookupToken }) => {
+  const decodedInvoice = decodeInvoice(paymentRequest)
+
+  if (decodedInvoice instanceof Error) return decodedInvoice
+
+  const { paymentHash, paymentSecret } = decodedInvoice
+
+  // TODO: Improve the following check with a non public payment secret
+  if (paymentSecret !== lookupToken) {
+    return new LnInvoiceLookupError("Invalid invoice data")
+  }
+
+  return {
+    invoiceIsPaid: async (): Promise<boolean | LnInvoiceLookupError> => {
+      const invoiceUser = await InvoiceUser.findOne({ _id: paymentHash })
```

Comment:
What if mongoose throws because there is no connection or something like that?

Reply:
Ideally, once we have a good entry point for mongoose operations, we can follow this pattern there and catch cases like these. 

We can of course return an error here directly but I think we need to move forward with the high priorities first before we nail every possible error case (that's not currently handled in the existing schema). Same thing goes for tests, let's convert the tests we currently have while we change the outer layer, then go another rounds focusing on coverage.